simplejson/simplejson.go

```diff
@@ -0,0 +1,268 @@
+package simplejson
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/v3io/frames"
+	"github.com/v3io/frames/pb"
+)
+
+type SimpleJsonRequestInterface interface {
+	CreateResponse(frames.FrameIterator) (interface{}, error)
+	ParseRequest([]byte) error
+	GetReadRequest() *frames.ReadRequest
+}
+
+type RequestSimpleJsonBase struct {
+	PanelId        int                      `json:"panelId"`
+	Range          map[string]interface{}   `json:"range"`
+	RangeRaw       map[string]string        `json:"rangeRaw"`
+	Interval       string                   `json:"interval"`
+	IntervalMs     int                      `json:"intervalMs"`
+	Targets        []map[string]interface{} `json:"targets"`
+	Target         string                   `json:"target"`
+	Format         string                   `json:"format"`
+	MaxDataPoints  int                      `json:"maxDataPoints"`
+	Path           string                   `json:"path"`
+	responseCreate SimpleJsonRequestInterface
+}
+
+type SimpleJsonQueryRequest struct {
+	RequestSimpleJsonBase
+	Filter    string
+	Fields    []string
+	TableName string
+	Type      string
+	Backend   string
+	From      string
+	To        string
+	Container string
+	Username  string
+	Password  string
+}
+
+type SimpleJsonSearchRequest struct {
+	SimpleJsonQueryRequest
+}
+
+type TableOutput struct {
+	Columns []map[string]string `json:"columns"`
+	Rows    [][]interface{}     `json:"rows"`
+	Type    string              `json:"type"`
+}
+
+type TimeSeriesOutput struct {
+	Datapoints [][]interface{} `json:"datapoints"`
+	Target     string          `json:"target"`
+}
+
+const QUERY_SEPARATOR = ";"
+const FIELDS_ITEMS_SEPARATOR = ","
+const DEFAULT_BACKEND = "tsdb"
+
+func SimpleJsonRequestFactory(method string, request []byte) (SimpleJsonRequestInterface, error) {
```

Comment:
IMO this is not required since in the handler you know which one you need.

Reply:
it's one way to have the read function get called always. I will make all these requests non-exported so the factory will be required. What say you?

Reply:
Don't see the value of not exporting these. I prefer simpler code every time :)